config/smartyfront.config.inc.php

```diff
@@ -171,8 +173,20 @@ function smartyTranslate($params, $smarty)
     if (!isset($params['sprintf'])) {
         $params['sprintf'] = array();
     }
+
+    // fix inheritance template filename in case of includes from different cross sources between theme, modules, ...
+    $filename = $smarty->template_resource;
+    if (!isset($smarty->inheritance->sourceStack[0]) || $filename === $smarty->inheritance->sourceStack[0]->resource) {
```

Comment:
You chould check the type of `sourceStack[0]` or if `resource` exists.

Reply:
if `$smarty->inheritance->sourceStack[0]` exists, `$smarty->inheritance->sourceStack[0]->resource` necessarily exists too.
Here you can see that what's pushed to `sourceStack`:
https://github.com/smarty-php/smarty/blob/master/libs/sysplugins/smarty_internal_runtime_inheritance.php#L246

It's a `Smarty_Template_Source`:
https://github.com/smarty-php/smarty/blob/master/libs/sysplugins/smarty_internal_template.php#L71

and `Smarty_Template_Source` contains `$resource`:
https://github.com/smarty-php/smarty/blob/master/libs/sysplugins/smarty_template_source.php#L25

Reply:
But like it's public properties, we can define any types so we should check the type of sourceStack[0]

Reply:
Ping @atomiix